source/val/ValidationState.cpp

```diff
@@ -251,20 +253,25 @@ vector<uint32_t> ValidationState_t::UnresolvedForwardIds() const {
 }
 
 bool ValidationState_t::IsDefinedId(uint32_t id) const {
-  return all_definitions_.find(Id{id}) != end(all_definitions_);
+  return all_definitions_.find(id) != end(all_definitions_);
 }
 
-const Id* ValidationState_t::FindDef(uint32_t id) const {
-  if (all_definitions_.count(Id{id}) == 0) {
+const Instruction* ValidationState_t::FindDef(uint32_t id) const {
+  if (all_definitions_.count(id) == 0) {
     return nullptr;
   } else {
     /// We are in a const function, so we cannot use defs.operator[]().
     /// Luckily we know the key exists, so defs_.at() won't throw an
     /// exception.
-    return &all_definitions_.at(id);
+    return all_definitions_.at(id);
   }
 }
 
+Instruction* ValidationState_t::FindDef(uint32_t id) {
+  return const_cast<Instruction*>(
```

Comment:
The const version of the  `FindDef` function is returning a const `Instruction` pointer. Because I need to track the usage of these instructions, I need to be able to modify the Instruction object. This function returns a non-const `Instruction` pointer which can be used for that purpose and since I didn't want to repeat code I took this approach(Item 3 of Effective C++).

Another approach would be to make the `uses_` object mutable or repeat code as you said. I thought this was the safest of the 3 options. I don't have a particularly strong opinion on the matter. I can remove the consts.